doc/srv6/SRv6_uSID.md

```diff
@@ -0,0 +1,112 @@
+# SONiC uSID
+
+## Table of Content
+- [Overview](#Overview)
+- [Scope](#Scope)
+- [Design](#Design)
+- [Example](#Example)
+
+## Revision
+
+| Rev  |   Date    |              Author              | Change Description      |
+| :--: | :-------: | :------------------------------: | :---------------------: |
+| 0.1  | 7/17/2022 | Shitanshu Shah, Reshma Sudarshan |  Initial version        |
+
+
+## Overview
+SRv6 uSID (micro-segment) is extension of the SRv6 network programming model [SRv6 uSID instructions IETF draft](https://datatracker.ietf.org/doc/draft-filsfils-spring-net-pgm-extension-srv6-usid/). uSID is a compressed SID value which can be for example carried in 16-bits (unlike full IPv6 address to represent a SID). uSID as is designed scales well with much lower MTU overhead required per uSID carrier. uSID carrier is 128-bit IPv6 address that can carry upto 6 uSIDs [Refer to Example for more details]
+
+## Scope
+The scope of this document is to enhance orchagent to support uSID programming instructions in this IETF draft. Current SAI API definitions already support uSID instructions. No SAI API change required in scope of this document. Current version of routing protocols in SONiC does not support SRv6, it is not in the scope of this document to add such a support for FRR routing stack.

```

Comment:
I think, currently we use FRR 8.1 version on SONiC which supports SRv6, any plan to use FRR stack for SRv6 feature?

Reply:
Yes, FRR SRv6 integration is being worked upon at this time. We will have an update on this soon.